pypit/armasters.py

```diff
@@ -74,75 +76,93 @@ def master_name(ftype, setup, mdir=None):
                      normpixelflat='{:s}/MasterFlatField_{:s}.fits'.format(mdir, setup),
                      arc='{:s}/MasterArc_{:s}.fits'.format(mdir, setup),
                      wave='{:s}/MasterWave_{:s}.fits'.format(mdir, setup),
-                     wave_calib='{:s}/MasterWaveCalib_{:s}.json'.format(mdir, setup),
+                     wv_calib='{:s}/MasterWaveCalib_{:s}.json'.format(mdir, setup),
                      tilts='{:s}/MasterTilts_{:s}.fits'.format(mdir, setup),
                      slitprof='{:s}/MasterSlitProfile_{:s}.fits'.format(mdir, setup),
                      sensfunc='{:s}/MasterSensFunc_{:s}_{:s}.yaml'.format(mdir, setup[0], setup[-2:]),
                      )
     return name_dict[ftype]
 
-'''
-def load_masters(slf, det, setup):
-    """ Load master frames
+def get_master_frame(slf, mftype, det=None):
+    """ If a MasterFrame exists, load it
+    And peform some extra steps for specific calibrations
     Parameters
     ----------
     slf
-    det
-    setup
+    mftype : str
+    det : int, optional
+
     Returns
     -------
+    msfile : ndarray or dict
+
     """
-    def load_master(file, exten=0):
-        hdu = pyfits.open(file)
-        data = hdu[exten].data
-        return data
-    # Bias
-    slf._msbias[det-1] = load_master(master_name('bias', setup))
-    # Bad Pixel
-    slf._bpix[det-1] = load_master(master_name('badpix', setup))
-    # Trace
-    slf._mstrace[det-1] = load_master(master_name('trace', setup))
-    slf._pixcen[det-1] = load_master(master_name('trace', setup), exten=1)
-    slf._pixwid[det-1] = load_master(master_name('trace', setup), exten=2)
-    slf._lordpix[det-1] = load_master(master_name('trace', setup), exten=3)
-    slf._rordpix[det-1] = load_master(master_name('trace', setup), exten=4)
-    # Flat
-    slf._mspixelflatnrm[det-1] = load_master(master_name('normpixelflat', setup))
-    # Arc/wave
-    slf._msarc[det-1] = load_master(master_name('arc', setup))
-    slf._mswave[det-1] = load_master(master_name('wave', setup))
-    # Tilts
-    slf._tilts[det-1] = load_master(master_name('tilts', setup))
-'''
 
+    setup = settings.argflag['reduce']['masters']['setup']
+    # Were MasterFrames even desired?
+    if (settings.argflag['reduce']['masters']['reuse']) or (settings.argflag['reduce']['masters']['force']):
+        ms_name = master_name(mftype, setup)
+        try:
+            msfile, head = arload.load_master(ms_name, frametype=mftype)
+        except IOError:
+            msgs.warn("No Master frame found of type {:s}: {:s}".format(mftype,ms_name))
+            raise IOError
```

Comment:
Well, in this case I actually wish to raise the IOError
and only throw the warning.

Am leaving as is.